core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -189,23 +277,23 @@ public String getDetailedInfo() {
     }
   }
 
-  static class BlockIterator<T> {
+  static class BlockBuffer<T> {
     /**
-     * Constructor to create a BlockIterator.
+     * Constructor to create a BlockBuffer.
      * @param filePath file path
      */
-    public BlockIterator(String filePath) {
+    public BlockBuffer(String filePath) {
     }
 
-    public List<T> getNextBatchBlocks() throws AlluxioRuntimeException {
+    public T getNextBatchBlocks() throws AlluxioRuntimeException {
```

Comment:
Why does it need to be generic? I also don't really see how this can return a generic when you actually implement this.